sdk/src/Core/RuntimeDependencyRegistry.cs

```diff
@@ -0,0 +1,166 @@
+        }
+
+        private void RegisterInstance(string assemblyName, string className, object instance)
+        {
+            try
+            {
+                _rwlock.EnterWriteLock();
+
+                _runtimeDependency[FormatKey(assemblyName, className)] = instance;
+            }
+            finally
+            {
+                if (_rwlock.IsWriteLockHeld)
+                {
+                    _rwlock.ExitWriteLock();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Get the registerd runtime dependency instance. This method is used by the SDK's core package and for service packages 
+        /// to get a reference to another service package service client.
+        /// </summary>
+        /// <typeparam name="T">The type that should be returned from the registry.</typeparam>
+        /// <param name="assemblyName"></param>
+        /// <param name="className"></param>
+        /// <returns></returns>
+        public T GetInstance<T>(string assemblyName, string className)
+            where T : class
+        {
+            try
+            {
+                _rwlock.EnterReadLock();
+
+                if (_runtimeDependency.TryGetValue(FormatKey(assemblyName, className), out object instance))
+                {
+                    return instance as T;
+                }
+
+                return null;
+            }
+            finally
+            {
+                if(_rwlock.IsReadLockHeld)
+                {
+                    _rwlock.ExitReadLock();
+                }
+            }
+        }
+
+        private string FormatKey(string assemblyName, string className) => $"{assemblyName}_{className}";
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposedValue)
+            {
+                if (disposing)
+                {
+                    _rwlock.Dispose();
+                }
+
+                disposedValue = true;
+            }
+        }
+
+        /// <summary>
+        /// Dispose the instance.
+        /// </summary>
+        public void Dispose()
+        {
+            // Do not change this code. Put cleanup code in 'Dispose(bool disposing)' method
+            Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
+    }
+
+    /// <summary>
+    /// This exception is thrown when the SDK is used as part of an application compiled for Native AOT with trimming turned on.
+    /// In this environment assemblies can not be dynamically loaded and the usual ServiceClientHelpers operations will fail.
+    /// </summary>
+    public class MissingRuntimeDependencyException : AmazonClientException
+    {
+
+        public MissingRuntimeDependencyException(string packageName, string className, string registerMethod)
+            : base("Operation failed because of missing runtime dependency. In Native AOT builds runtime dependencies can not be " +
```

Comment:
`because of missing` => `because of a missing`

Reply:
Fixed